2-WebApp-graph-user/2-1-Call-MSGraph/Startup.cs

```diff
@@ -36,8 +37,7 @@ public void ConfigureServices(IServiceCollection services)
             // and chosen token cache implementation
             services.AddAzureAdV2Authentication(Configuration)
                 .AddMsal(new string[] { Constants.ScopeUserRead })
-                .AddInMemoryTokenCache()
-                    ;
+                .AddInMemoryAppTokenCache();
```

Comment:
2-1 changed because Microsoft.Identity.Web changed
replaced this  with AddInMemoryPerUserTokenCache()

---
In reply to: [267319301](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/65#discussion_r267319301) [](ancestors = 267319301)

Reply:
I get that, but why don't you use `AddInMemoryPerUserTokenCache` @kalyankrishna1 ? this is a per user token cache here?

Reply:
Also you probably want to change 3 as well?

Reply:
You are right. All confidential client scenarios that have a call to Graph or Other Apis should have both App and user caches

---
In reply to: [268398295](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/65#discussion_r268398295) [](ancestors = 268398295)

Reply:
3?

---
In reply to: [268398338](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/65#discussion_r268398338) [](ancestors = 268398338)

Reply:
in theory yes, for the app token cache. In practice only when you use client creds